src/Microsoft.AspNet.Authentication.Facebook/Notifications/FacebookAuthenticationNotifications.cs

```diff
@@ -23,16 +24,33 @@ public FacebookAuthenticationNotifications()
         /// <summary>
         /// Gets or sets the function that is invoked when the Authenticated method is invoked.
         /// </summary>
-        public Func<FacebookAuthenticatedContext, Task> OnAuthenticated { get; set; }
+        public new Func<FacebookAuthenticatedContext, Task> OnAuthenticated { get; set; }
```

Comment:
So my gut feeling is that we should try to find some way to share some of this mechanical code that's in all of the social providers, the notifications class for example seems like it would benefit from a generic base class so you don't have to do these ugly new's to make it return the right context.